camel-idea-plugin/src/main/java/org/apache/camel/idea/facet/CamelFacetConfiguration.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.idea.facet;
+
+import javax.swing.*;
+import com.intellij.facet.FacetConfiguration;
+import com.intellij.facet.ui.FacetEditorContext;
+import com.intellij.facet.ui.FacetEditorTab;
+import com.intellij.facet.ui.FacetValidatorsManager;
+import com.intellij.openapi.util.InvalidDataException;
+import com.intellij.openapi.util.WriteExternalException;
+import org.jdom.Element;
+import org.jetbrains.annotations.Nls;
+import org.jetbrains.annotations.NotNull;
+
+public class CamelFacetConfiguration implements FacetConfiguration {
+    @Override
+    public FacetEditorTab[] createEditorTabs(FacetEditorContext facetEditorContext, FacetValidatorsManager facetValidatorsManager) {
+        return new FacetEditorTab[]{new FacetEditorTab() {
+            @NotNull
+            @Override
+            public JComponent createComponent() {
+                return new JLabel("Content goes here");
```

Comment:
@adrianbumbas yeah lets research a bit what IDEA offers and what is best. I wanted to have a window etc that scans your project and detect all your Camel routes, and list them, so you can see them in one window, and then click on the route to open the file in the source editor and position the cursor on the line where the route starts.

I guess facet is not the answer. If facet is better for project configuration then for Camel that makes less sense. If you use Camel with Spring Boot, the the SB facet is for configuring SB and Camel can do that too with SB configuration.